pkg/plugin/trans.go

```diff
@@ -21,8 +21,12 @@ import (
 	"github.com/neondatabase/autoscaling/pkg/util"
 )
 
+// resourceTransition maintains the current state of its resource and handles the transition
+// into a new state.
 type resourceTransition[T constraints.Unsigned] struct {
-	node    *nodeResourceState[T]
+	// node represents the current state of the node
+	node *nodeResourceState[T]
+	// oldNode is the old state of the node, which we just keep for convenience
```

Comment:
let's split this into a separate PR? afaict, the changes aren't related to the rest of this PR, and don't need to be held up on the review on the rest of it